examples/src/components/CopyLink.tsx

```diff
@@ -1,17 +1,21 @@
 'use client'
 import { ArrowUpRightIcon } from '@heroicons/react/24/outline'
+import { usePathname, useSearchParams } from 'next/navigation'
 
 import Title from '@/components/Title'
 
 export default function CopyLink() {
+  const pathname = usePathname()
+  const params = useSearchParams()
+
   return (
-    <div className="mr-2 mb-2 text-left items-center text-neutral-500 border-2 border-yellow-200 rounded-lg bg-yellow-50 px-6 py-3">
+    <div className="mb-2 text-left items-center text-neutral-500 border-2 border-yellow-200 rounded-lg bg-yellow-50 px-6 py-3">
       <div className="flex justify-between items-center pb-1">
         <Title>Collaborate on this document</Title>
         <div className="flex">
           <a
             className="text-sm flex items-center gap-1 px-3 py-1 rounded-lg bg-pink-950 text-white border transition-all "
-            href={window.location.href}
+            href={`${pathname}?doc=${params.get('doc')}`}
```

Comment:
Strange, but this doesn't appear to be working for me. When I click the button, it opens a new window with the right path, but the query string is `doc=null`. 🤔

Reply:
Ahhh - it looks like this bug might be because [shared layouts aren't re-rendered after a navigation](https://nextjs.org/docs/app/building-your-application/routing/linking-and-navigating#3-partial-rendering). So whatever the search params were when the user landed on the site is what they'll be stuck as :-(